packages/pn-commons/src/utils/apikeys.utility.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box } from '@mui/material';
+import { ReactNode } from 'react';
+import { useTranslation } from 'react-i18next';
+import { getLocalizedOrDefaultLabel } from '../services/localization.service';
+import { ApiKeyStatus, ApiKeyStatusHistory } from '../types/ApiKeys';
+
+function localizeStatus(
+  status: string,
+  defaultLabel: string,
+  defaultDescription: string,
+  history: Array<ApiKeyStatusHistory>
+): {
+  label: string;
+  tooltip: ReactNode;
+  description: string;
+} {
+  return {
+    label: getLocalizedOrDefaultLabel('apikeys', `status.${status}`, defaultLabel),
+    tooltip: TooltipApiKey(history),
+    description: getLocalizedOrDefaultLabel(
+      'apikeys',
+      `status.${status}-description`,
+      defaultDescription
+    ),
+  };
+}
+
+const TooltipApiKey = (history: Array<ApiKeyStatusHistory>) => {
+  const { t } = useTranslation(['apikeys']);
+  return (
+    <Box
+      sx={{
+        '> .MuiBox-root:not(:last-child)': {
+          marginBottom: 1,
+        },
+      }}
+    >
+      {history.map((h) => {
```

Comment:
@NicolaGiornetta in questo switch: separare la presentazione dal messaggio da mostrare. il messaggio può essere salvato in una variabile separata che viene renderizzata all'interno del componente desiderato, evitando duplicazione di codice non necessaria. Andiamo a tipizzare anche h con ApiStatusHistory